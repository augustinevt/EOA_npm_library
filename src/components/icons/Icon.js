import React from 'react';
import propTypes from 'prop-types';
import icons from './iconSvgBank';

class Icon extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const type = this.props.type;
    const icon = icons[type] ? icons[type] : icons['column'];
    const styles = {overflow: 'hidden'};
    const width = this.props.width;
    const height = this.props.height;

    if (width) {
      styles.width = width;
    } else {
      styles.height = height;
    }

    return (
      <svg className="icon border"
        viewBox={ icon.viewBox }
        style={ styles }
        version="1.1"
        xmlns="http://www.w3.org/2000/svg">
          { icon.svg }
      </svg>
    );
  }
}

Icon.defaultProps = { type: 'column' };
Icon.propTypes = {
  width: propTypes.number,
  height: propTypes.number,
  type: propTypes.string,
};

export default Icon;
