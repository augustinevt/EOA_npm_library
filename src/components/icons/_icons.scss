.icon-wrapper {
  text-align: center;
  svg {
    width: 40px;
    height: 40px;
  }
}
