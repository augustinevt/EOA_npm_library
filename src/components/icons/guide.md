# Icon Reference

```
  <Icon
    type="icon"
    height={20}
    width={30}
  />
```
### props:
- type: string
- height: number (optional)
- width: number (optional)

### TODO
  - ### Replace height and width with size prop
