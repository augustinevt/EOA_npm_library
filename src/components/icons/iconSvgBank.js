import React from 'react';

const bank = {
  stream: {
    viewBox: '0 0 2048 1792',
    svg: (
      <g>
          <path d="M1152,1152v128H512v-128H1152z M1152,384v640H512V384H1152z M1792,1152v128h-512v-128H1792z M1792,896v128h-512V896H1792z M1792,640v128h-512V640H1792z M1792,384v128h-512V384H1792z M256,1344V384H128v960c0,17.3,6.3,32.3,19,45s27.7,19,45,19s32.3-6.3,45-19S256,1361.3,256,1344z M1920,1344V256H384v1088c0,22-3.7,43.3-11,64h1483c17.3,0,32.3-6.3,45-19 S1920,1361.3,1920,1344z M2048,128v1216c0,53.3-18.7,98.7-56,136s-82.7,56-136,56H192c-53.3,0-98.7-18.7-136-56s-56-82.7-56-136V256 h256V128H2048z"/>
      </g>
    )
  },

  orders: {
    viewBox: '0 0 1792 1792',
    svg: (
      <g>
        <path d="M704 1536q0 52-38 90t-90 38-90-38-38-90 38-90 90-38 90 38 38 90zm896 0q0 52-38 90t-90 38-90-38-38-90 38-90 90-38 90 38 38 90zm128-1088v512q0 24-16.5 42.5t-40.5 21.5l-1044 122q13 60 13 70 0 16-24 64h920q26 0 45 19t19 45-19 45-45 19h-1024q-26 0-45-19t-19-45q0-11 8-31.5t16-36 21.5-40 15.5-29.5l-177-823h-204q-26 0-45-19t-19-45 19-45 45-19h256q16 0 28.5 6.5t19.5 15.5 13 24.5 8 26 5.5 29.5 4.5 26h1201q26 0 45 19t19 45z"/>
      </g>
    )
  },

  watchlist: {
    viewBox: '0 0 1792 1792',
    svg: (
      <g>
        <path d="M704 320v768q0 26-19 45t-45 19v576q0 26-19 45t-45 19h-512q-26 0-45-19t-19-45v-512l249-873q7-23 31-23h424zm320 0v704h-256v-704h256zm768 896v512q0 26-19 45t-45 19h-512q-26 0-45-19t-19-45v-576q-26 0-45-19t-19-45v-768h424q24 0 31 23zm-1056-1184v224h-352v-224q0-14 9-23t23-9h288q14 0 23 9t9 23zm672 0v224h-352v-224q0-14 9-23t23-9h288q14 0 23 9t9 23z"/>
      </g>
    )
  },

  inventory: {
    viewBox: '0 0 1792 1792',
    svg: (
      <g>
        <path d="M402 707l494 305-342 285-490-319zm986 555v108l-490 293v1l-1-1-1 1v-1l-489-293v-108l147 96 342-284v-2l1 1 1-1v2l343 284zm-834-1144l342 285-494 304-338-270zm836 589l338 271-489 319-343-285zm-151-589l489 319-338 270-494-304z"/>
      </g>
    )
  },

  trends: {
    viewBox: '0 0 2048 1792',
    svg: (
      <g>
        <path d="M2048 1536v128h-2048v-1536h128v1408h1920zm-128-1248v435q0 21-19.5 29.5t-35.5-7.5l-121-121-633 633q-10 10-23 10t-23-10l-233-233-416 416-192-192 585-585q10-10 23-10t23 10l233 233 464-464-121-121q-16-16-7.5-35.5t29.5-19.5h435q14 0 23 9t9 23z"/>
      </g>
    )
  },

  profit: {
    viewBox: '0 0 1792 1792',
    svg: (
      <g>
        <path d="M1362 1185q0 153-99.5 263.5t-258.5 136.5v175q0 14-9 23t-23 9h-135q-13 0-22.5-9.5t-9.5-22.5v-175q-66-9-127.5-31t-101.5-44.5-74-48-46.5-37.5-17.5-18q-17-21-2-41l103-135q7-10 23-12 15-2 24 9l2 2q113 99 243 125 37 8 74 8 81 0 142.5-43t61.5-122q0-28-15-53t-33.5-42-58.5-37.5-66-32-80-32.5q-39-16-61.5-25t-61.5-26.5-62.5-31-56.5-35.5-53.5-42.5-43.5-49-35.5-58-21-66.5-8.5-78q0-138 98-242t255-134v-180q0-13 9.5-22.5t22.5-9.5h135q14 0 23 9t9 23v176q57 6 110.5 23t87 33.5 63.5 37.5 39 29 15 14q17 18 5 38l-81 146q-8 15-23 16-14 3-27-7-3-3-14.5-12t-39-26.5-58.5-32-74.5-26-85.5-11.5q-95 0-155 43t-60 111q0 26 8.5 48t29.5 41.5 39.5 33 56 31 60.5 27 70 27.5q53 20 81 31.5t76 35 75.5 42.5 62 50 53 63.5 31.5 76.5 13 94z"/>
      </g>
    )
  },

  settings: {
    viewBox: '0 0 1792 1792',
    svg: (
      <g>
        <path d="M480 1408v128h-352v-128h352zm352-128q26 0 45 19t19 45v256q0 26-19 45t-45 19h-256q-26 0-45-19t-19-45v-256q0-26 19-45t45-19h256zm160-384v128h-864v-128h864zm-640-512v128h-224v-128h224zm1312 1024v128h-736v-128h736zm-960-1152q26 0 45 19t19 45v256q0 26-19 45t-45 19h-256q-26 0-45-19t-19-45v-256q0-26 19-45t45-19h256zm640 512q26 0 45 19t19 45v256q0 26-19 45t-45 19h-256q-26 0-45-19t-19-45v-256q0-26 19-45t45-19h256zm320 128v128h-224v-128h224zm0-512v128h-864v-128h864z"/>
      </g>
    )
  },

  dashboard: {
    viewBox: '0 0 1792 1792',
    svg: (
      <g>
        <path d="M384 1152q0-53-37.5-90.5t-90.5-37.5-90.5 37.5-37.5 90.5 37.5 90.5 90.5 37.5 90.5-37.5 37.5-90.5zm192-448q0-53-37.5-90.5t-90.5-37.5-90.5 37.5-37.5 90.5 37.5 90.5 90.5 37.5 90.5-37.5 37.5-90.5zm428 481l101-382q6-26-7.5-48.5t-38.5-29.5-48 6.5-30 39.5l-101 382q-60 5-107 43.5t-63 98.5q-20 77 20 146t117 89 146-20 89-117q16-60-6-117t-72-91zm660-33q0-53-37.5-90.5t-90.5-37.5-90.5 37.5-37.5 90.5 37.5 90.5 90.5 37.5 90.5-37.5 37.5-90.5zm-640-640q0-53-37.5-90.5t-90.5-37.5-90.5 37.5-37.5 90.5 37.5 90.5 90.5 37.5 90.5-37.5 37.5-90.5zm448 192q0-53-37.5-90.5t-90.5-37.5-90.5 37.5-37.5 90.5 37.5 90.5 90.5 37.5 90.5-37.5 37.5-90.5zm320 448q0 261-141 483-19 29-54 29h-1402q-35 0-54-29-141-221-141-483 0-182 71-348t191-286 286-191 348-71 348 71 286 191 191 286 71 348z"/>
      </g>
    )
  },

  user: {
    viewBox: '0 0 1792 1792',
    svg: (
      <g>
        <path d="M896 0q182 0 348 71t286 191 191 286 71 348q0 181-70.5 347t-190.5 286-286 191.5-349 71.5-349-71-285.5-191.5-190.5-286-71-347.5 71-348 191-286 286-191 348-71zm619 1351q149-205 149-455 0-156-61-298t-164-245-245-164-298-61-298 61-245 164-164 245-61 298q0 250 149 455 66-327 306-327 131 128 313 128t313-128q240 0 306 327zm-235-647q0-159-112.5-271.5t-271.5-112.5-271.5 112.5-112.5 271.5 112.5 271.5 271.5 112.5 271.5-112.5 112.5-271.5z"/>
      </g>
    )
  },

  trash: {
    viewBox: '0 0 1792 1792',
    svg: (
      <g>
        <path d="M704 736v576q0 14-9 23t-23 9h-64q-14 0-23-9t-9-23v-576q0-14 9-23t23-9h64q14 0 23 9t9 23zm256 0v576q0 14-9 23t-23 9h-64q-14 0-23-9t-9-23v-576q0-14 9-23t23-9h64q14 0 23 9t9 23zm256 0v576q0 14-9 23t-23 9h-64q-14 0-23-9t-9-23v-576q0-14 9-23t23-9h64q14 0 23 9t9 23zm128 724v-948h-896v948q0 22 7 40.5t14.5 27 10.5 8.5h832q3 0 10.5-8.5t14.5-27 7-40.5zm-672-1076h448l-48-117q-7-9-17-11h-317q-10 2-17 11zm928 32v64q0 14-9 23t-23 9h-96v948q0 83-47 143.5t-113 60.5h-832q-66 0-113-58.5t-47-141.5v-952h-96q-14 0-23-9t-9-23v-64q0-14 9-23t23-9h309l70-167q15-37 54-63t79-26h320q40 0 79 26t54 63l70 167h309q14 0 23 9t9 23z"/>
      </g>
    )
  },

  submit: {
    viewBox: '0 0 1792 1792',
    svg: (
      <g>
        <path d="M1363 877l-742 742q-19 19-45 19t-45-19l-166-166q-19-19-19-45t19-45l531-531-531-531q-19-19-19-45t19-45l166-166q19-19 45-19t45 19l742 742q19 19 19 45t-19 45z"/>
      </g>
    )
  },

  ellipsis: {
    viewBox: '0 0 1792 1792',
    svg: (
      <g>
        <path d="M576 736v192q0 40-28 68t-68 28h-192q-40 0-68-28t-28-68v-192q0-40 28-68t68-28h192q40 0 68 28t28 68zm512 0v192q0 40-28 68t-68 28h-192q-40 0-68-28t-28-68v-192q0-40 28-68t68-28h192q40 0 68 28t28 68zm512 0v192q0 40-28 68t-68 28h-192q-40 0-68-28t-28-68v-192q0-40 28-68t68-28h192q40 0 68 28t28 68z"/>
      </g>
    )
  },

  download: {
    viewBox: '0 0 2048 1792',
    svg: (
      <g>
        <path d="M1344 928q0-14-9-23t-23-9h-224v-352q0-13-9.5-22.5t-22.5-9.5h-192q-13 0-22.5 9.5t-9.5 22.5v352h-224q-13 0-22.5 9.5t-9.5 22.5q0 14 9 23l352 352q9 9 23 9t23-9l351-351q10-12 10-24zm640 224q0 159-112.5 271.5t-271.5 112.5h-1088q-185 0-316.5-131.5t-131.5-316.5q0-130 70-240t188-165q-2-30-2-43 0-212 150-362t362-150q156 0 285.5 87t188.5 231q71-62 166-62 106 0 181 75t75 181q0 76-41 138 130 31 213.5 135.5t83.5 238.5z"/>
      </g>
    )
  },

  growth: {
    viewBox: '0 0 1792 1792',
    svg: (
      <g>
        <path d="M522 883q-74-162-74-371h-256v96q0 78 94.5 162t235.5 113zm1078-275v-96h-256q0 209-74 371 141-29 235.5-113t94.5-162zm128-128v128q0 71-41.5 143t-112 130-173 97.5-215.5 44.5q-42 54-95 95-38 34-52.5 72.5t-14.5 89.5q0 54 30.5 91t97.5 37q75 0 133.5 45.5t58.5 114.5v64q0 14-9 23t-23 9h-832q-14 0-23-9t-9-23v-64q0-69 58.5-114.5t133.5-45.5q67 0 97.5-37t30.5-91q0-51-14.5-89.5t-52.5-72.5q-53-41-95-95-113-5-215.5-44.5t-173-97.5-112-130-41.5-143v-128q0-40 28-68t68-28h288v-96q0-66 47-113t113-47h576q66 0 113 47t47 113v96h288q40 0 68 28t28 68z"/>
      </g>
    )
  },

  loading: {
    viewBox: '0 0 40 40',
    svg: (
    <g>
        <path opacity="0.2" d="M20.201,5.169c-8.254,0-14.946,6.692-14.946,14.946c0,8.255,6.692,14.946,14.946,14.946s14.946-6.691,14.946-14.946C35.146,11.861,28.455,5.169,20.201,5.169z M20.201,31.749c-6.425,0-11.634-5.208-11.634-11.634c0-6.425,5.209-11.634,11.634-11.634c6.425,0,11.633,5.209,11.633,11.634C31.834,26.541,26.626,31.749,20.201,31.749z"/>
       <path d="M26.013,10.047l1.654-2.866c-2.198-1.272-4.743-2.012-7.466-2.012h0v3.312h0C22.32,8.481,24.301,9.057,26.013,10.047z">
       <animateTransform attributeType="xml"
         attributeName="transform"
         type="rotate"
         from="0 20 20"
         to="360 20 20"
         dur="0.5s"
         repeatCount="indefinite"/>
       </path>
    </g>
    ),
  },
  image: {
    viewBox: '0 0 1024 896',
    svg: (
      <g>
        <path d="M1024,576V896H0V704L192,512,320,640,704,256ZM192,384q-80,0-136-56T0,192Q0,112,56,56T192,0q80,0,136,56t56,136q0,80-56,136T192,384Z"/>
      </g>
    )
  },
};

export default bank;
