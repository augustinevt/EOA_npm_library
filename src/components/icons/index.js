import Icon from './Icon';
import iconSvgBank from './iconSvgBank';

export { iconSvgBank as iconSvgBank };
export default Icon;
