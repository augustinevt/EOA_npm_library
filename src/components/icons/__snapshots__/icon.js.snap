// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard icon 1`] = `
<div
  className="icon-wrapper"
>
  <svg
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
      }
    }
    version="1.1"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M384 1152q0-53-37.5-90.5t-90.5-37.5-90.5 37.5-37.5 90.5 37.5 90.5 90.5 37.5 90.5-37.5 37.5-90.5zm192-448q0-53-37.5-90.5t-90.5-37.5-90.5 37.5-37.5 90.5 37.5 90.5 90.5 37.5 90.5-37.5 37.5-90.5zm428 481l101-382q6-26-7.5-48.5t-38.5-29.5-48 6.5-30 39.5l-101 382q-60 5-107 43.5t-63 98.5q-20 77 20 146t117 89 146-20 89-117q16-60-6-117t-72-91zm660-33q0-53-37.5-90.5t-90.5-37.5-90.5 37.5-37.5 90.5 37.5 90.5 90.5 37.5 90.5-37.5 37.5-90.5zm-640-640q0-53-37.5-90.5t-90.5-37.5-90.5 37.5-37.5 90.5 37.5 90.5 90.5 37.5 90.5-37.5 37.5-90.5zm448 192q0-53-37.5-90.5t-90.5-37.5-90.5 37.5-37.5 90.5 37.5 90.5 90.5 37.5 90.5-37.5 37.5-90.5zm320 448q0 261-141 483-19 29-54 29h-1402q-35 0-54-29-141-221-141-483 0-182 71-348t191-286 286-191 348-71 348 71 286 191 191 286 71 348z"
      />
    </g>
  </svg>
</div>
`;

exports[`download icon 1`] = `
<div
  className="icon-wrapper"
>
  <svg
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
      }
    }
    version="1.1"
    viewBox="0 0 2048 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M1344 928q0-14-9-23t-23-9h-224v-352q0-13-9.5-22.5t-22.5-9.5h-192q-13 0-22.5 9.5t-9.5 22.5v352h-224q-13 0-22.5 9.5t-9.5 22.5q0 14 9 23l352 352q9 9 23 9t23-9l351-351q10-12 10-24zm640 224q0 159-112.5 271.5t-271.5 112.5h-1088q-185 0-316.5-131.5t-131.5-316.5q0-130 70-240t188-165q-2-30-2-43 0-212 150-362t362-150q156 0 285.5 87t188.5 231q71-62 166-62 106 0 181 75t75 181q0 76-41 138 130 31 213.5 135.5t83.5 238.5z"
      />
    </g>
  </svg>
</div>
`;

exports[`ellipsis icon 1`] = `
<div
  className="icon-wrapper"
>
  <svg
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
      }
    }
    version="1.1"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M576 736v192q0 40-28 68t-68 28h-192q-40 0-68-28t-28-68v-192q0-40 28-68t68-28h192q40 0 68 28t28 68zm512 0v192q0 40-28 68t-68 28h-192q-40 0-68-28t-28-68v-192q0-40 28-68t68-28h192q40 0 68 28t28 68zm512 0v192q0 40-28 68t-68 28h-192q-40 0-68-28t-28-68v-192q0-40 28-68t68-28h192q40 0 68 28t28 68z"
      />
    </g>
  </svg>
</div>
`;

exports[`inventory icon 1`] = `
<div
  className="icon-wrapper"
>
  <svg
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
      }
    }
    version="1.1"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M402 707l494 305-342 285-490-319zm986 555v108l-490 293v1l-1-1-1 1v-1l-489-293v-108l147 96 342-284v-2l1 1 1-1v2l343 284zm-834-1144l342 285-494 304-338-270zm836 589l338 271-489 319-343-285zm-151-589l489 319-338 270-494-304z"
      />
    </g>
  </svg>
</div>
`;

exports[`orders icon 1`] = `
<div
  className="icon-wrapper"
>
  <svg
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
      }
    }
    version="1.1"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M704 1536q0 52-38 90t-90 38-90-38-38-90 38-90 90-38 90 38 38 90zm896 0q0 52-38 90t-90 38-90-38-38-90 38-90 90-38 90 38 38 90zm128-1088v512q0 24-16.5 42.5t-40.5 21.5l-1044 122q13 60 13 70 0 16-24 64h920q26 0 45 19t19 45-19 45-45 19h-1024q-26 0-45-19t-19-45q0-11 8-31.5t16-36 21.5-40 15.5-29.5l-177-823h-204q-26 0-45-19t-19-45 19-45 45-19h256q16 0 28.5 6.5t19.5 15.5 13 24.5 8 26 5.5 29.5 4.5 26h1201q26 0 45 19t19 45z"
      />
    </g>
  </svg>
</div>
`;

exports[`profit icon 1`] = `
<div
  className="icon-wrapper"
>
  <svg
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
      }
    }
    version="1.1"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M1362 1185q0 153-99.5 263.5t-258.5 136.5v175q0 14-9 23t-23 9h-135q-13 0-22.5-9.5t-9.5-22.5v-175q-66-9-127.5-31t-101.5-44.5-74-48-46.5-37.5-17.5-18q-17-21-2-41l103-135q7-10 23-12 15-2 24 9l2 2q113 99 243 125 37 8 74 8 81 0 142.5-43t61.5-122q0-28-15-53t-33.5-42-58.5-37.5-66-32-80-32.5q-39-16-61.5-25t-61.5-26.5-62.5-31-56.5-35.5-53.5-42.5-43.5-49-35.5-58-21-66.5-8.5-78q0-138 98-242t255-134v-180q0-13 9.5-22.5t22.5-9.5h135q14 0 23 9t9 23v176q57 6 110.5 23t87 33.5 63.5 37.5 39 29 15 14q17 18 5 38l-81 146q-8 15-23 16-14 3-27-7-3-3-14.5-12t-39-26.5-58.5-32-74.5-26-85.5-11.5q-95 0-155 43t-60 111q0 26 8.5 48t29.5 41.5 39.5 33 56 31 60.5 27 70 27.5q53 20 81 31.5t76 35 75.5 42.5 62 50 53 63.5 31.5 76.5 13 94z"
      />
    </g>
  </svg>
</div>
`;

exports[`settings icon 1`] = `
<div
  className="icon-wrapper"
>
  <svg
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
      }
    }
    version="1.1"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M480 1408v128h-352v-128h352zm352-128q26 0 45 19t19 45v256q0 26-19 45t-45 19h-256q-26 0-45-19t-19-45v-256q0-26 19-45t45-19h256zm160-384v128h-864v-128h864zm-640-512v128h-224v-128h224zm1312 1024v128h-736v-128h736zm-960-1152q26 0 45 19t19 45v256q0 26-19 45t-45 19h-256q-26 0-45-19t-19-45v-256q0-26 19-45t45-19h256zm640 512q26 0 45 19t19 45v256q0 26-19 45t-45 19h-256q-26 0-45-19t-19-45v-256q0-26 19-45t45-19h256zm320 128v128h-224v-128h224zm0-512v128h-864v-128h864z"
      />
    </g>
  </svg>
</div>
`;

exports[`stream icon 1`] = `
<div
  className="icon-wrapper"
>
  <svg
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
      }
    }
    version="1.1"
    viewBox="0 0 2048 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M1152,1152v128H512v-128H1152z M1152,384v640H512V384H1152z M1792,1152v128h-512v-128H1792z M1792,896v128h-512V896H1792z M1792,640v128h-512V640H1792z M1792,384v128h-512V384H1792z M256,1344V384H128v960c0,17.3,6.3,32.3,19,45s27.7,19,45,19s32.3-6.3,45-19S256,1361.3,256,1344z M1920,1344V256H384v1088c0,22-3.7,43.3-11,64h1483c17.3,0,32.3-6.3,45-19 S1920,1361.3,1920,1344z M2048,128v1216c0,53.3-18.7,98.7-56,136s-82.7,56-136,56H192c-53.3,0-98.7-18.7-136-56s-56-82.7-56-136V256 h256V128H2048z"
      />
    </g>
  </svg>
</div>
`;

exports[`submit icon 1`] = `
<div
  className="icon-wrapper"
>
  <svg
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
      }
    }
    version="1.1"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M1363 877l-742 742q-19 19-45 19t-45-19l-166-166q-19-19-19-45t19-45l531-531-531-531q-19-19-19-45t19-45l166-166q19-19 45-19t45 19l742 742q19 19 19 45t-19 45z"
      />
    </g>
  </svg>
</div>
`;

exports[`trash icon 1`] = `
<div
  className="icon-wrapper"
>
  <svg
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
      }
    }
    version="1.1"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M704 736v576q0 14-9 23t-23 9h-64q-14 0-23-9t-9-23v-576q0-14 9-23t23-9h64q14 0 23 9t9 23zm256 0v576q0 14-9 23t-23 9h-64q-14 0-23-9t-9-23v-576q0-14 9-23t23-9h64q14 0 23 9t9 23zm256 0v576q0 14-9 23t-23 9h-64q-14 0-23-9t-9-23v-576q0-14 9-23t23-9h64q14 0 23 9t9 23zm128 724v-948h-896v948q0 22 7 40.5t14.5 27 10.5 8.5h832q3 0 10.5-8.5t14.5-27 7-40.5zm-672-1076h448l-48-117q-7-9-17-11h-317q-10 2-17 11zm928 32v64q0 14-9 23t-23 9h-96v948q0 83-47 143.5t-113 60.5h-832q-66 0-113-58.5t-47-141.5v-952h-96q-14 0-23-9t-9-23v-64q0-14 9-23t23-9h309l70-167q15-37 54-63t79-26h320q40 0 79 26t54 63l70 167h309q14 0 23 9t9 23z"
      />
    </g>
  </svg>
</div>
`;

exports[`trends icon 1`] = `
<div
  className="icon-wrapper"
>
  <svg
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
      }
    }
    version="1.1"
    viewBox="0 0 2048 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M2048 1536v128h-2048v-1536h128v1408h1920zm-128-1248v435q0 21-19.5 29.5t-35.5-7.5l-121-121-633 633q-10 10-23 10t-23-10l-233-233-416 416-192-192 585-585q10-10 23-10t23 10l233 233 464-464-121-121q-16-16-7.5-35.5t29.5-19.5h435q14 0 23 9t9 23z"
      />
    </g>
  </svg>
</div>
`;

exports[`user icon 1`] = `
<div
  className="icon-wrapper"
>
  <svg
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
      }
    }
    version="1.1"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M896 0q182 0 348 71t286 191 191 286 71 348q0 181-70.5 347t-190.5 286-286 191.5-349 71.5-349-71-285.5-191.5-190.5-286-71-347.5 71-348 191-286 286-191 348-71zm619 1351q149-205 149-455 0-156-61-298t-164-245-245-164-298-61-298 61-245 164-164 245-61 298q0 250 149 455 66-327 306-327 131 128 313 128t313-128q240 0 306 327zm-235-647q0-159-112.5-271.5t-271.5-112.5-271.5 112.5-112.5 271.5 112.5 271.5 271.5 112.5 271.5-112.5 112.5-271.5z"
      />
    </g>
  </svg>
</div>
`;

exports[`watchlist icon 1`] = `
<div
  className="icon-wrapper"
>
  <svg
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
      }
    }
    version="1.1"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g>
      <path
        d="M704 320v768q0 26-19 45t-45 19v576q0 26-19 45t-45 19h-512q-26 0-45-19t-19-45v-512l249-873q7-23 31-23h424zm320 0v704h-256v-704h256zm768 896v512q0 26-19 45t-45 19h-512q-26 0-45-19t-19-45v-576q-26 0-45-19t-19-45v-768h424q24 0 31 23zm-1056-1184v224h-352v-224q0-14 9-23t23-9h288q14 0 23 9t9 23zm672 0v224h-352v-224q0-14 9-23t23-9h288q14 0 23 9t9 23z"
      />
    </g>
  </svg>
</div>
`;
